portal-web/docroot/html/taglib/ui/page_iterator/start.jsp

```diff
@@ -410,6 +412,67 @@ NumberFormat numberFormat = NumberFormat.getNumberInstance(locale);
 	}
 </script>
 
+<script type="text/javascript">
+
+	var dropdown = document.querySelector('.pagination-items-per-page');
+	var button = dropdown.querySelector('.dropdown-toggle');
+	var list = dropdown.querySelector('.dropdown-menu');
+	var options = list.querySelectorAll('.dropdown-item');
+	var selectedItemValue = button.dataset.attribute;
+
+	button.addEventListener('keydown', function(event) {
```

Comment:
As we have Senna.js implementation in DXP that tries to imitate a SPA, we need to make sure to remove the listener when navigating to the next page to avoid memory link issues. You can add at the end of the script to remove these listeners that were added, for example:

```js
var destroyDropDownPagination = function () {
	button.removeEventListener('keydown', onButtonKeyDown);

	Liferay.detach('beforeScreenFlip', destroyDropDownPagination);
};

Liferay.on('beforeScreenFlip', destroyDropDownPagination);
```

Remember, saving the function is a variable because to remove a listener you need the function reference in memory to remove.

```jsx
function onButtonKeyDown(event) {
	//...
}

button.addEventListener('keydown', onButtonKeyDown);

var destroyDropDownPagination = function () {
	button.removeEventListener('keydown', onButtonKeyDown);
	// ...

	Liferay.detach('beforeScreenFlip', destroyDropDownPagination);
};

Liferay.on('beforeScreenFlip', destroyDropDownPagination);
```

Reply:
Fixed